spec/lib/sidekiq/honeycomb_middleware_spec.rb

```diff
@@ -0,0 +1,57 @@
+require "rails_helper"
+
+class TestSidekiqWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: :low_priority
+
+  def perform(arg = nil)
+    raise StandardError, "BOOM" if arg == "fail"
+  end
+end
+
+RSpec.describe Sidekiq::HoneycombMiddleware do
```

Comment:
I would add a test of what happens if the argument is neither "fail" nor "dont fail" because I accidentally did and it opened a whole new world:

```ruby
  1) Sidekiq::HoneycombMiddleware sends an event with expected keys
     Failure/Error: expect(collected_data).to include(expected_hash)

       expected {"duration_ms" => 0.037999998312443495, "meta.beeline_version" => "1.3.0", "meta.local_hostname" => "groove..." => "fe0937e8-0a82-493d-9560-8b2e724d9555", "trace.trace_id" => "a0e02cfc-c134-448b-a140-275a3f07da41"} to include {"sidekiq.args" => ["dont fail"]}
       Diff:


       @@ -1,6 +1,14 @@
       -"sidekiq.args" => ["dont fail"],
       +"duration_ms" => 0.037999998312443495,
       +"meta.beeline_version" => "1.3.0",
       +"meta.local_hostname" => "groove.local",
       +"meta.span_type" => "root",
       +"name" => "sidekiq.job",
       +"service_name" => nil,
       +"sidekiq.args" => ["x"],
        "sidekiq.class" => "TestSidekiqWorker",
       -"sidekiq.jid" => an_instance_of(String),
       +"sidekiq.jid" => "e2042599741e98aaf3ef7197",
        "sidekiq.queue" => "low_priority",
        "sidekiq.result" => "success",
       +"trace.span_id" => "fe0937e8-0a82-493d-9560-8b2e724d9555",
       +"trace.trace_id" => "a0e02cfc-c134-448b-a140-275a3f07da41",
```

Reply:
this was more of a event detection issue, which should be fixed now :)

Reply:
this was more of a event detection issue, which should be fixed now :)